core/src/main/java/org/apache/carbondata/core/indexstore/ExtendedBlocklet.java

```diff
@@ -168,7 +169,7 @@ public void serializeData(DataOutput out, Map<String, Short> uniqueLocation, boo
       throws IOException {
     super.write(out);
     if (isCountJob) {
-      out.writeLong(inputSplit.getRowCount());
+      out.writeLong(inputSplit.getIndexRow().getInt(BlockletIndexRowIndexes.ROW_COUNT_INDEX));
```

Comment:
Please add below comment on separate commit, I will merge this.

In CarbonInputSplit, getDetailInfo() is a lazy call. we want to avoid this during count(*). As rowCount is filled inside getDetailInfo(). In count(*) case we may not have proper row count. So, always take row count from indexRow.

Reply:
done